### DISCUSSION TITLE
Orin AGX Pinmux not been applied

### DISCUSSION THREAD

**Discussion by 0nlytom** [OE4T/meta-tegra#1591] [Q&A] [answered]
Hi, 
I working on the custom broad and I'm having issues with applying the `pinmux` and `padvoltage` files. 
When flashing the device, both the `pinmux` and `padvoltage` files are not applied, however the `tegra-devicetree` is applied. 
The files tree is as followings 

> ├── tegra-binaries
> │   ├── tegra-bootfiles
> │   │   ├── tegra234-eac-5000-gpio-default.dtsi
> │   │   ├── tegra234-eac-5000-padvoltage-default.dtsi
> │   │   └── tegra234-eac-5000-pinmux.dtsi
> │   ├── tegra-bootfiles_35.5.0.bbappend
> │   ├── tegra-flashvars
> │   │   └── eac-5000
> │   │       └── flashvars
> │   ├── tegra-flashvars_35.5.0.bbappend
> └── tegra-devicetree
>     ├── tegra-devicetree
>     │   └── tegra234-p3701-0000-p3737-0000-eac-5000.dts
>     └── tegra-devicetree_1.0.bb

Contents of 'flashvars' is: here

[flashvars.txt](https://github.com/user-attachments/files/15929143/flashvars.txt)

Everything else is directly taken from [Creating-a-custom-MACHINE ](https://github.com/OE4T/meta-tegra/wiki/Creating-a-custom-MACHINE )
Flashing logs are here. 
[flash_log.txt](https://github.com/user-attachments/files/15929153/flash_log.txt)

Where have i gone wrong? This is tested with [tegra-demo](https://github.com/OE4T/tegra-demo-distro) too.

**Comment by madisongh** [comment]
Which branch are you using?  The mechanism for customizing the flashvars changed between kirkstone and scarthgap/scarthgap-l4t-r35.x, and the wiki page hasn't yet been updated to reflect that.  The new process doesn't use a bbappend for the `tegra-flashvars` recipe - the contents of the flashvars file gets generated from `TEGRA_FLASHVAR_...` variables that you set in your machine configuration.

If you are still on kirkstone, then it looks like your replacement flashvars file didn't get picked up - compare the copy you added to your layer against the copy that's in the tegraflash package - and troubleshooting your bbappend for the `tegra-flashvars` recipe would be the right next step.

**Reply by 0nlytom** [reply]
All working as excepted. Thank you.

**Reply by 0nlytom** [reply]
Not sure what is going on, but something is not going right during the flashing of the Orin AGX. I'm currently experiencing the same issue as [here](https://forums.developer.nvidia.com/t/rgmii-for-agx-orin/286486/8). 
Could you please have a look at [flash.log](https://github.com/user-attachments/files/15967500/flash.log) and see if there is anything going wrong. 
There is the custom bootfiles that I'm using [tegra234-eac-5000-.zip](https://github.com/user-attachments/files/15967517/tegra234-eac-5000-.zip).

Thank you

**Reply by madisongh** [reply]
Did you figure this out? The flash log looks OK. I took a quick look at that forum thread, and it sounds more like there could be a problem with what's in your device tree, rather than a problem with the flashing process.

**Reply by 0nlytom** [reply, marked as answer]
Yes, thank you, I managed to resolve this a few days ago and forgot to update it here. It was an issue with few GPIO's that were been changed during booting up by an mismatch in the device tree of the kernel.

**Comment by 0nlytom** [comment]
Using the scarthgap branch. This makes sense then as I tested it, however, the files where not found. Could you please tell me where to put the pinmux and padvoltage files?

**Comment by spb06** [comment]
Hey there, just wondering if this ended up working?I am going through something very similar involving a custom carrier board (AGX Orin).

We followed the instructions in the "[Create a Custom Machine](https://github.com/OE4T/meta-tegra/wiki/Creating-a-custom-MACHINE#custom-device-tree)". Specifically, we created a new machine config and flash config as advised in the wiki.

For the machine conf file, we copied over the [jetson-agx-orin-devkit.conf](https://github.com/OE4T/meta-tegra/blob/master/conf/machine/jetson-agx-orin-devkit.conf) and renamed it to match our machine name. The carrier board vendor provided their own .dtb file that we believe should replace the .dtb file specified in [jetson-agx-orin-devkit.conf](https://github.com/OE4T/meta-tegra/blob/master/conf/machine/jetson-agx-orin-devkit.conf) but we are not sure the best way to go about doing that.

Any guidance will be greatly appreciated!
